roles/openshift_version/tasks/main.yml

```diff
@@ -119,30 +127,42 @@
 
 - fail:
     msg: openshift_version role was unable to set openshift_version
```

Comment:
They are all supposed to be set by the role. The inventory can set `openshift_image_tag` and `openshift_pkg_version` which are described in https://github.com/openshift/openshift-ansible/blob/master/inventory/byo/hosts.ose.example#L35 but they are only generally needed if the role can't figure it out some other way. `openshift_version` really shouldn't be set by the user at all, though nothing stops them and some inventories probably do.

You're right, though, having some advice would be useful. It would be good to know if there are any likely situations where the user might hit these so we could address them.